### PR TITLE
Increase version to 0.24

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Review by elichai — APPROVED** [review]
:D

**Review by TheBlueMatt — APPROVED** [review]
I didn't review the changelog, but I've been testing against master for a bit and things are working for us.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,18 @@
...
+- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/416) `Transaction::get_size` method
+- [Export](https://github.com/rust-bitcoin/rust-bitcoin/pull/412) `util::amount::Denomination`
+- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/417) `Block::get_size` and `Block::get_weight` methods
+- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/415) `MerkleBlock::from_header_txids`
+- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/429) `BlockHeader::u256_from_compact_target`
+- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/448) `feefilter` network message
+- [Cleanup/replace](https://github.com/rust-bitcoin/rust-bitcoin/pull/397) `Script::Instructions` iterator API
+- [Disallow uncompressed pubkeys in witness address generation](https://github.com/rust-bitcoin/rust-bitcoin/pull/428)
+- [Deprecate](https://github.com/rust-bitcoin/rust-bitcoin/pull/451) `util::contracthash` module
+
```
Comment (dr-orlovsky) [inline CHANGELOG.md:15]:
```suggestion
- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/435) modulo division operation for `Uint128` and `Uint256`
- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/436) `slice_to_u64_be` endian conversion method
- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/463) `Clone` to `RawNetworkMessage`

```

Comment (dr-orlovsky) [reply CHANGELOG.md:15]: - [Remove](https://github.com/rust-bitcoin/rust-bitcoin/pull/385) `BitcoinHash` trait
- [Add](https://github.com/rust-bitcoin/rust-bitcoin/pull/385) `Block::block_hash()` and `BlockHeader::block_hash()` methods

Comment (apoelstra) [reply CHANGELOG.md:15]: Sure

Comment (apoelstra) [reply CHANGELOG.md:15]: l'll give you the new methods, but I don't think new `#derive`s are worth including in the changelog

**Review by elichai — APPROVED** [review]
reACK

**Review by TheBlueMatt — APPROVED** [review]

**Pull request merged**